sdk/core/core/internal/az_span_internal.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_SPAN_INTERNAL_H
```

Comment:
that's entirely possible. However in this case the function I added should be in "internal" since it would be used SDK wide and not just in core. Whether the functionality in "private" should remain there is maybe for another PR or GH issue